STM32/Filtering/Src/array-math.c

```diff
@@ -103,24 +107,78 @@ void cbPush(double_cbuf_handle_t p_cb, double new_val) {
     }
 }
 
+/*!
+** @brief Get the newest element of the circular buffer
+*/
+double cbGetHead(double_cbuf_handle_t p_cb) {
+    if (p_cb->idx == 0)
+    {
+        return p_cb->buffer[p_cb->len - 1];
+    }
+    return p_cb->buffer[p_cb->idx - 1];
+}
+
+/*!
+** @brief Get the oldest element of the circular buffer
+*/
+double cbGetTail(double_cbuf_handle_t p_cb) {
+    return p_cb->buffer[p_cb->idx];
+}
+
 /*!
 ** @brief Computes the moving average of an array in a circular fashion.
 */
-double maMean(moving_avg_cbuf_handle_t p_ma, double newVal)
+double maMean(moving_avg_cbuf_handle_t p_ma, double new_val)
 {
     // Update sum of array
-    p_ma->sum = p_ma->sum - p_ma->cbuf_t.buffer[p_ma->cbuf_t.idx] + newVal;
+    p_ma->sum = p_ma->sum - p_ma->cbuf_t.buffer[p_ma->cbuf_t.idx] + new_val;
     // Update value in array
-    p_ma->cbuf_t.buffer[p_ma->cbuf_t.idx] = newVal;    
-    // Update idx
-    p_ma->cbuf_t.idx++;
-    if (p_ma->cbuf_t.idx >= p_ma->cbuf_t.len)
-    {
-        p_ma->cbuf_t.idx = 0;
-    }
+    p_ma->cbuf_t.buffer[p_ma->cbuf_t.idx] = new_val;    
+
+    // Push new value onto buffer
+    cbPush(&p_ma->cbuf_t, new_val);
```

Comment:
Aren't the previous two lines of code doing the same thing?

Reply:
Yes they are missed that.